cmd/command.go

```diff
@@ -20,36 +23,61 @@ func NewRootCommand() *cobra.Command {
 		},
 	}
 
-	cmd.PersistentFlags().StringVarP(
-		&app.configPath, "config-path", "c", "./merger.yaml",
-		"Path to the configuration file.")
-	cmd.PersistentFlags().IntVar(
-		&app.port, "listen-port", 8080,
-		"Listen port for the HTTP server.")
+	app.Bind(cmd)
 
 	cmd.AddCommand(NewVersionCommand())
 
 	return cmd
 }
 
 type App struct {
-	configPath string
-	port       int
+	viper *viper.Viper
 }
 
-func (app *App) run(cmd *cobra.Command, args []string) {
-	config, err := ReadConfig(app.configPath)
-	if err != nil {
-		log.WithField("error", err).Error("failed to load config")
-		return
-	}
+func (app *App) Bind(cmd *cobra.Command) {
+	app.viper = viper.New()
+	app.viper.SetEnvPrefix("MERGER")
+	app.viper.AutomaticEnv()
+
+	configPath := cmd.PersistentFlags().StringP(
+		"config-path", "c", "",
+		"Path to the configuration file.")
+	cobra.OnInitialize(func() {
+		if configPath != nil && *configPath != "" {
+			config, err := ReadConfig(*configPath)
+			if err != nil {
+				log.WithField("error", err).Errorf("failed to load config file '%s'", *configPath)
+				os.Exit(1)
+				return
+			}
 
+			urls := []string{}
+			for _, e := range config.Exporters {
+				urls = append(urls, e.URL)
+			}
+			app.viper.SetDefault("urls", strings.Join(urls, " "))
+		}
+	})
+
+	cmd.PersistentFlags().Int(
+		"listen-port", 8080,
+		"Listen port for the HTTP server. (ENV:MERGER_PORT)")
```

Comment:
MERGER prefix set in viper, shouldn't this string be formatted with it instead of hardcode?  Don't mind if that's not trivial.

Reply:
I did not find any method to get the variable out of Viper again. We could just use the same string we give to Viper, but it is altering it (ie making it uppercase).

Reply:
Don't mind then:)